tests/Sandpit/Exscientia/Process/test_openmm.py

```diff
@@ -1,38 +1,75 @@
-import BioSimSpace.Sandpit.Exscientia as BSS
+import math
+import re
 
+import BioSimSpace.Sandpit.Exscientia as BSS
 import pytest
 
 # Store the tutorial URL.
 url = BSS.tutorialUrl()
 
+# Store the allowed restraints.
+restraints = BSS.Protocol._position_restraint._PositionRestraintMixin.restraints()
 
 @pytest.fixture(scope="session")
 def system():
     """Re-use the same molecuar system for each test."""
     return BSS.IO.readMolecules(["tests/input/ala.top", "tests/input/ala.crd"])
 
 
-def test_minimise(system):
+@pytest.mark.parametrize("restraint", restraints)
+def test_minimise(system, restraint, file_regression, request):
     """Test a minimisation protocol."""
 
     # Create a short minimisation protocol.
-    protocol = BSS.Protocol.Minimisation(steps=100)
+    protocol = BSS.Protocol.Minimisation(
+        steps=100, restraint=restraint, force_constant=1000
+        )
 
     # Run the process, check that it finished without error, and returns a system.
-    run_process(system, protocol)
+    new_system = run_process(
+        system, protocol, file_regression=file_regression, request=request
+        )
+
+    if restraint == "none":
+        return
+
+    # Check if restrained atoms stayed in place
+    restraint_atoms = system.getRestraintAtoms(restraint)
+    did_not_move = [
+        get_dist_atoms(system.getAtom(x), new_system.getAtom(x)) < 0.05
+        for x in restraint_atoms
+    ]
 
+    assert all(did_not_move)
 
-@pytest.mark.parametrize("restraint", ["backbone", "heavy", "all", "none"])
-def test_equilibrate(system, restraint):
+
+@pytest.mark.parametrize("restraint", restraints)
+def test_equilibrate(system, restraint, file_regression, request):
     """Test an equilibration protocol."""
 
     # Create a short equilibration protocol.
     protocol = BSS.Protocol.Equilibration(
-        runtime=BSS.Types.Time(0.001, "nanoseconds"), restraint=restraint
+        runtime=BSS.Types.Time(0.001, "nanoseconds"),
+        restraint=restraint,
+        force_constant=1000
     )
 
     # Run the process, check that it finished without error, and returns a system.
-    run_process(system, protocol)
+    new_system = run_process(
+        system, protocol, file_regression, request
+        )
+
+    if restraint == "none":
```

Comment:
Alright, I disagree about file regression because I think it is cleaner than testing explicitly for certain lines and easier to use when you don't know the exact content of the line before you actually ran the test. But it's your call. I removed the file regression bit now.